File: Initial Prototype/ServerSide/FindNDrive/FindNDriveServices2/DTOs/RegisterDTO.cs
// --------------------------------------------------------------------------------------------------------------------
// <copyright file="RegisterDTO.cs" company="">
//   
// </copyright>
// <summary>
//   DataContract that will be sent as part of the IUserService.RegisterUser request
// </summary>
// --------------------------------------------------------------------------------------------------------------------

namespace FindNDriveServices2.DTOs
{
    using System.ComponentModel.DataAnnotations;
    using System.Runtime.Serialization;

    using DomainObjects.Domains;

    using FindNDriveInfrastructureCore;

    /// <summary>
    /// DataContract that will be sent as part of the IUserService.RegisterUser request
    /// </summary>
    [DataContract]
    public class RegisterDTO
    {
        /// <summary>
        /// Gets or sets the user.
        /// </summary>
        [DataMember]
        [Required]
        [ValidateObject]
        public User User { get; set; }

        /// <summary>
        /// Gets or sets the password.
        /// </summary>
        [DataMember]
        [Required]
        [Compare("ConfirmedPassword", ErrorMessage = "The password and confirmation password do not match.")]
        public string Password { get; set; }

        /// <summary>
        /// Gets or sets the confirmed password.
        /// </summary>
        [DataMember]
        [Required]
        [Compare("Password", ErrorMessage = "The password and confirmation password do not match.")]
        public string ConfirmedPassword { get; set; }
    }
}

File: Initial Prototype/Client/MyApplicationProject/MyApplication/src/main/java/com/example/myapplication/DTOs/RegisterDTO.java
package com.example.myapplication.dtos;

import com.example.myapplication.domain_objects.User;

/**
 * Created by Michal on 05/11/13.
 */
public class RegisterDTO {
    private User User;
    private String Password;
    private String ConfirmedPassword;
    public RegisterDTO(String password, String confirmedPassword, User user) {
        ConfirmedPassword = confirmedPassword;
        Password = password;
        User = user;
    }
}
